tfsdk/plan.go

```diff
@@ -163,6 +172,105 @@ func (p *Plan) SetAttribute(ctx context.Context, path *tftypes.AttributePath, va
 	return diags
 }
 
+// pathExists walks the current state and returns true if the path can be reached.
+// The value at the path may be null or unknown.
+func (p Plan) pathExists(ctx context.Context, path *tftypes.AttributePath) (bool, diag.Diagnostics) {
```

Comment:
Is my understanding correct that this is the same method on Plan and State?

Reply:
Yes. Is there more to this question or a suggestion for how to improve it? The implementation felt small enough with the release time pressure that didn't feel warranted to refactor this further immediately, but can certainly spend cycles to update this further.

I could see that refactoring potentially going multiple ways since we don't own the `Plan.Raw`/`State.Raw` type (`tftypes.Value`), for example:

1. We could create an abstraction type in the framework and create a method on that. Then code becomes `p.Raw.pathExists(ctx, path)`. Seems related to https://github.com/hashicorp/terraform-plugin-framework/issues/172.
1. We could create a shared function that takes in that type as a parameter. Then code becomes `pathExists(p.Raw, ctx, path)`.
1. We could create a shared type that `Config`, `Plan`, and `State` all implement for their "data handling", which owns the method.